go/tasks/pluginmachinery/catalog/reader_array_processor.go

```diff
@@ -0,0 +1,131 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/ioutils"
+	"github.com/lyft/flytestdlib/bitarray"
+
+	"github.com/lyft/flyteplugins/go/tasks/errors"
+
+	"github.com/lyft/flytestdlib/logger"
+
+	"github.com/lyft/flyteplugins/go/tasks/pluginmachinery/workqueue"
+)
+
+type ArrayReaderWorkItem struct {
+	// ArrayReaderWorkItem outputs:
+	cached   *bitarray.BitSet
+	progress *bitarray.BitSet
+
+	// ArrayReaderWorkItem Inputs:
+	downloadRequest DownloadArrayRequest
+}
+
+func (item ArrayReaderWorkItem) CachedResults() *bitarray.BitSet {
+	return item.cached
+}
+
+func NewArrayReaderWorkItem(request DownloadArrayRequest) *ArrayReaderWorkItem {
+	return &ArrayReaderWorkItem{
+		downloadRequest: request,
+	}
+}
+
+type ArrayReaderProcessor struct {
+	catalogClient    Client
+	maxItemsPerRound int
+}
+
+func (p ArrayReaderProcessor) Process(ctx context.Context, workItem workqueue.WorkItem) (workqueue.WorkStatus, error) {
+	wi, casted := workItem.(*ArrayReaderWorkItem)
+	if !casted {
+		return workqueue.WorkStatusNotDone, fmt.Errorf("wrong work item type. Received: %v", reflect.TypeOf(workItem))
+	}
+
+	if wi.cached == nil {
+		wi.cached = bitarray.NewBitSet(uint(wi.downloadRequest.Count))
+		wi.progress = bitarray.NewBitSet(uint(wi.downloadRequest.Count))
+	}
+
+	isArray := wi.downloadRequest.Count > 0
```

Comment:
Can you add a comment for this?  Don't understand.  We still want the loop to run even if the Count is 0, why?  When is the count 0?